File: src/my_py_pkg/my_py_pkg/hardware_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus

class HardwareStatusPublisher(Node):
  def __init__(self):
    super().__init__("hardware_status_publisher")
    self.create_timer(1.0, self.timer_callback)
    self.hardware_status_publisher_ = self.create_publisher(HardwareStatus, "hardware_status", 10)
    self.get_logger().info("HardwareStatusPublisher node has been started")

  def timer_callback(self):
    self.publish_hardware_status()
  
  def publish_hardware_status(self):
    msg = HardwareStatus()
    msg.temperature = 71.2
    msg.are_motors_ready = True
    msg.debug_message = "Nothing special"
    self.hardware_status_publisher_.publish(msg)

def main(args=None):
  rclpy.init(args=args)
  node = HardwareStatusPublisher()
  rclpy.spin(node)
  rclpy.shutdown()


if __name__ == "__main__":
  main()

File: src/my_cpp_pkg/src/hardware_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

using namespace std::chrono_literals;

class HardwareStatusPublisher : public rclcpp::Node
{
public:
  HardwareStatusPublisher() : Node("hardware_status_publisher")
  {
    hardware_status_publisher_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>(
        "hardware_status", 10);
    timer_ = this->create_wall_timer(
      1s, std::bind(&HardwareStatusPublisher::timerCallback, this));
    RCLCPP_INFO(this->get_logger(), "HardwareStatusPublisher node has been started");
  }

private:
  void timerCallback()
  {
    this->publishHardwareStatus();
  }

  void publishHardwareStatus()
  {
    auto msg = my_robot_interfaces::msg::HardwareStatus();
    msg.temperature = 68.4;
    msg.are_motors_ready = false;
    msg.debug_message = "Motors are cooling off";
    hardware_status_publisher_->publish(msg);
  }

  rclcpp::TimerBase::SharedPtr timer_;
  rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr hardware_status_publisher_;

};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<HardwareStatusPublisher>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}